### ISSUE TITLE
Bad ergonomics when upgrading iOS Simulator

### ISSUE THREAD

**Issue by LinusU** [expo/expo-cli#1775]
## Description

Since SDK 37 was released today I decided to upgrade my project. When doing this the tool asked me if I wanted to upgrade the iOS Simulator. Doing that sounded good so I selected yes. At this point the iOS simulator wasn't running.

## Expected Behavior

I expected the iOS Simulator to be "upgraded" so that I was ready to rock with SDK 37 🚀 

## Observed Behavior

I got this error:

```text
iOS Simulator device failed to boot. Try opening Simulator first, then running your app.
Timed out waiting for iOS Simulator device to boot.
```

Okay. No worries. I'll just launch the simulator myself and then re-run `expo upgrade`. Said and done, this happened the second time:

```text
Error running `xcrun simctl boot C0BF453B-94B1-4136-B360-DDC641919291`: An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=165):
Unable to boot device in current state: Booted

There was a problem booting a device in iOS Simulator. Quit Simulator, and try again.
xcrun exited with non-zero code: 165
```

Wait? You want me to _quit_ the simulator now? But I just started it for you, since you told me to!

-----

I'm guessing that the fix here would be to check if the simulator is running before trying to boot it, maybe? Or check for "Unable to boot device in current state: Booted" and see that that means that it's already booted, and thus the failure can be ignored?

## Environment

```text
  Expo CLI 3.17.5 environment info:
    System:
      OS: macOS 10.15.4
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.12.0 - /usr/local/bin/node
      Yarn: 1.22.4 - ~/my-project/node_modules/.bin/yarn
      npm: 6.13.6 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.4/11E146 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^37.0.0 => 37.0.3 
      react: 16.9.0 => 16.9.0 
      react-native: https://github.com/expo/react-native/archive/sdk-37.0.0.tar.gz => 0.61.4 
    npmGlobalPackages:
      expo-cli: 3.13.1 (I was running expo-cli via npx, so this version wasn't used)
```

## Reproducible Demo

not sure how 😅

**Comment by brentvatne** [comment]
I think there isn't so much that we can do about making this more reliable, we use the APIs that Apple provides :\ We can maybe provide some helpful messages though.

Also, maybe we can remove this from the `upgrade` command or move it to the end, and if it fails we don't block on that, we just say to try running `expo client:install:ios` manually.

**Comment by LinusU** [comment]
Good suggestions 👍 

What do you think about this part:

> Or check for "Unable to boot device in current state: Booted" and see that that means that it's already booted, and thus the failure can be ignored?

**Comment by brentvatne** [comment]
Possibly! But it may be erroring because it's booted and still not responding. 😱

**Comment by brentvatne** [comment]
If you can repro this reliably on your machine is there any chance you'd be willing to investigate your suggestion and see if that helps?

**Comment by LinusU** [comment]
Unfortunately I haven't been able to reproduce it reliably. I added the following lines to `_bootDefaultSimulatorDeviceAsync`:

```diff
     return await _xcrunAsync(['simctl', 'boot', deviceUDID]);
   } catch (e) {
+    if (e.stderr.includes('Unable to boot device in current state: Booted')) {
+      console.error('Used the workaround!')
+      return
+    }
+
     _Logger().default.global.error(`There was a problem booting a device in iOS Simulator. Quit Simulator, and try again.`);

     throw e;
```

Which at one point made the command go thru and printing "Used the workaround!", but it's hard to tell if actually worked after that 🤔 at least the program didn't print any more errors 😅 

The actual error seems to happen quite rarely and I can only try and close the simulator and then have the script start it, and then sometimes that fails. When that fails there is a small chance that I will get the other error the next time I run the script...

**Comment by EvanBacon** [comment]
I fixed this error in #2541 along with a bunch of other simulator issues
```
Error running `xcrun simctl boot C0BF453B-94B1-4136-B360-DDC641919291`: An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=165):
Unable to boot device in current state: Booted

There was a problem booting a device in iOS Simulator. Quit Simulator, and try again.
xcrun exited with non-zero code: 165
```
